### DISCUSSION TITLE
Newbie: Opening the ChEBI database in SDF format

### DISCUSSION THREAD

**Discussion by edmondop** [rdkit/rdkit#4351] [Q&A]
Hello everyone,

I am trying to use the PandasTools to load the ChEBI complete sdf like the following

`frame = PandasTools.LoadSDF('datasources/chebi/ChEBI_complete.sdf')`

However, this triggers a variety of errors like so:
```

   "[15:07:45] WARNING: not removing hydrogen atom without neighbors\n",
      "[15:07:45] Explicit valence for atom # 12 N, 4, is greater than permitted\n",
      "[15:07:45] ERROR: Could not sanitize molecule ending on line 684160\n",
      "[15:07:45] ERROR: Explicit valence for atom # 12 N, 4, is greater than permitted\n",
      "[15:07:45] Explicit valence for atom # 12 N, 4, is greater than permitted\n",
      "[15:07:45] ERROR: Could not sanitize molecule ending on line 684428\n",
      "[15:07:45] ERROR: Explicit valence for atom # 12 N, 4, is greater than permitted\n",
      "[15:07:46] WARNING: not removing hydrogen atom with dummy atom neighbors\n",
      "[15:07:46] WARNING: not removing hydrogen atom with dummy atom neighbors\n",
      "[15:07:46] WARNING: not removing hydrogen atom with dummy atom neighbors\n",
      "[15:07:46] Can't kekulize mol.  Unkekulized atoms: 1 3 5 7 8\n",
      "\n",
      "[15:07:46] ERROR: Could not sanitize molecule ending on line 3010225\n",
      "[15:07:46] ERROR: Can't kekulize mol.  Unkekulized atoms: 1 3 5 7 8\n",
      "\n",
      "[15:07:47] Explicit valence for atom # 3 O, 3, is greater than permitted\n",
      "[15:07:47] ERROR: Could not sanitize molecule ending on line 3230072\n",
      "[15:07:47] ERROR: Explicit valence for atom # 3 O, 3, is greater than permitted\n",
      "[15:07:47] Explicit valence for atom # 12 N, 4, is greater than permitted\n",
      "[15:07:47] ERROR: Could not sanitize molecule ending on line 3369474\n",
      "[15:07:47] ERROR: Explicit valence for atom # 12 N, 4, is greater than permitted\n",
      "[15:07:47] \n",
      "\n",
      "****\n",
      "Post-condition Violation\n",
      "Element 'hv' not found\n",
      "Violation occurred on line 91 in file /private/var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T/pip-req-build-e2w56r9d/build/temp.macosx-10.9-x86_64-3.8/rdkit/rdkit/Code/GraphMol/PeriodicTable.h\n",
      "Failed Expression: anum > -1\n",
      "****\n",
      "\n",
      "[15:07:47] ERROR: Element 'hv' not found\n",
      "[15:07:47] ERROR: moving to the beginning of the next molecule\n",
```

I am a newbie in dealing with SDF files, is this a common / expected behaviour?
Thanks

**Comment by greglandrum** [comment]
Hi @edmondo1984,
The RDKit will generate errors like that for molecules where it has problems making sense of the chemistry. This happens during the "sanitization" process when a molecule is read: https://www.rdkit.org/docs/RDKit_Book.html#molecular-sanitization

**Reply by edmondop** [reply]
Thank you. Assuming the data source is correct since it's a public dataset, should I disable sanitisation?

**Reply by greglandrum** [reply]
I would never assume that a data source is correct just because it's a public dataset. Even a well-known public dataset. 

ChEBI, for example, includes things like this:
![image](https://user-images.githubusercontent.com/540511/127108191-3222311d-b52e-425b-b409-b4f7e4716381.png)
Which would be a reasonable molecule if a couple of the N-Fe bonds were marked as coordinate bonds, or if you know in advance that they should be converted to coordinate bonds, but which doesn't make chemical sense as is.

I would suggest looking at the molecules which are failing and seeing if you believe that they are chemically correct. If you think they are chemically correct, then go ahead and turn off sanitization for those molecules. If you don't think they are correct, you can either fix the structures or remove them (easiest done by just dropping molecules which the RDKit fails to sanitize).

If you don't have a chemistry background and don't feel like you know enough chemistry to be able to assess whether or not a particular structure is valid, then I would recommend dropping the structures which fail sanitization. The alternative, accepting every structure, is equivalent to saying: "I don't care about the quality of the data I'm working with".
